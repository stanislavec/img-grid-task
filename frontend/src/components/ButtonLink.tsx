import React from 'react';
import cn from 'classnames';
import { Link, LinkProps, useLocation } from 'react-router-dom';

import scss from './ButtonLink.module.scss';

type IButtonLink = LinkProps;

function ButtonLink(props: IButtonLink) {
  const { children, ...rest } = props;
  const { pathname } = useLocation();

  return (
    <Link {...rest} className={cn(scss.button, { [scss.active]: rest.to === pathname })}>
      {children}
    </Link>
  );
}

export default ButtonLink;
