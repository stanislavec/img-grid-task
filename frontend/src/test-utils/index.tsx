import React, { ReactElement } from 'react';
import { createBrowserRouter, createMemoryRouter, RouterProvider } from 'react-router-dom';

export const withRouter = (Component: ReactElement<unknown, any>, path: string = '/') => {
  const router = createBrowserRouter([
    {
      path,
      element: Component,
    },
  ]);

  return <RouterProvider router={router} />;
};

export const withMemotyRouter = (
  Component: ReactElement<unknown, any>,
  { route = '/', path = '/' } = {},
) => {
  const router = createMemoryRouter(
    [
      {
        path,
        element: Component,
      },
    ],
    {
      initialEntries: [route],
    },
  );
  return <RouterProvider router={router} />;
};
