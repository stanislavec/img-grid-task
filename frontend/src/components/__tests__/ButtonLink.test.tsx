import { render, screen } from '@testing-library/react';
import ButtonLink from '../ButtonLink';
import { withRouter } from '../../test-utils';

describe('ButtonLink', () => {
  it('is defined', () => {
    const elem = render(withRouter(<ButtonLink to="/">Defined link button</ButtonLink>));
    expect(elem.baseElement).toBeDefined();
  });
  it('has active class on current route', () => {
    const text = 'Link btn';
    const { rerender, baseElement } = render(withRouter(<ButtonLink to="/">{text}</ButtonLink>));

    expect(baseElement.querySelector('a')).toHaveClass('active');

    rerender(withRouter(<ButtonLink to="/somepath">{text}</ButtonLink>));

    const link = screen.getByText(text);

    if (!link) throw Error;

    expect(link.classList.contains('active')).toBe(false);
  });
  it('matches snapshot', () => {
    const elem = render(withRouter(<ButtonLink to="/">Link button</ButtonLink>));
    const link = elem.container.querySelector('a');
    expect(link).toMatchSnapshot();
  });
});
