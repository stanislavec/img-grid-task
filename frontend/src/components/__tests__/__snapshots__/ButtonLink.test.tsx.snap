// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonLink matches snapshot 1`] = `
<a
  class="button active"
  href="/"
>
  Link button
</a>
`;
